src/main/java/me/zeroeightsix/kami/module/modules/chat/AntiSpam.java

```diff
@@ -75,55 +75,61 @@
                 .collect(Collectors.toList())
                 .forEach(entry -> messageHistory.remove(entry.getKey()));
 
-        if (detectSpam(removeUsername(sPacketChat.getChatComponent().getUnformattedText()))) event.cancel();
+        if (isSpam(sPacketChat.getChatComponent().getUnformattedText())) event.cancel();
     });
 
-    private String removeUsername(String username) { return username.replaceAll("<[^>]*> ", ""); }
-
     @Override
     public void onEnable() { messageHistory = new ConcurrentHashMap<>(); }
 
     @Override
     public void onDisable() { messageHistory = null; }
 
-    private boolean detectSpam(String message) {
-        if (!filterOwn.getValue() && findPatterns(FilterPatterns.OWN_MESSAGE, message)) return false;
+    private boolean isSpam(String message) {
+        if (!filterOwn.getValue() && findPatterns(FilterPatterns.OWN_MESSAGE, message, false)) {
+            return false;
+        } else {
+            return detectSpam(removeUsername(message));
+        }
+    }
 
-        if (wordsLongerThen.getValue() && findPatterns(FilterPatterns.LONG_WORD, message)) return sendResult(wordsLongerThen.getName(), message);
+    private String removeUsername(String username) { return username.replaceAll("<[^>]*> ", ""); }
 
-        if (greenText.getValue() && findPatterns(FilterPatterns.GREEN_TEXT, message)) return sendResult(greenText.getName(), message);
+    private boolean detectSpam(String message) {
```

Comment:
What the hell am I witnessing. What on earth. Just add all of these to a map or something and loop over it!